reactor-core/src/main/java/reactor/core/publisher/FluxBufferPredicate.java

```diff
@@ -148,6 +148,7 @@ public void request(long n) {
 					s.request(Long.MAX_VALUE);
 				}
 				else {
+					long r = REQUESTED.get(this);
```

Comment:
@simonbasle is it possible that two Threads can execute this line at the same time? if so probably it should be handled

Reply:
@robotmrv a `Subscriber` is responsible for ensuring the upstream requests it makes are not done in parallel (rule [2.7](https://github.com/reactive-streams/reactive-streams-jvm#2.7)) so that line MUST NOT be executed at the same time. on the other hand, this operator needs to make sure both `s.request(...)` calls made in `emit` and `request` are serialized...